core/src/main/scala/gigahorse/Stream.scala

```diff
@@ -31,7 +31,11 @@ abstract class Stream[A] {
   def foreach(f: A => Unit): Future[Unit]
 
   /** Runs f on each element received to the stream with its previous output. */
-  def fold[B](zero: B)(f: (B, A) => B): Future[B]
+  final def fold[B](zero: B)(f: (B, A) => B): Future[B] =
+    foldResource(zero)(f, () => ())
```

Comment:
I think it makes `Stream` more useful if could keep this free of implementation.

Reply:
Thanks for the feedback. I reverted `Stream`'s fold implementation for flexibility, and moved it to concrete classes.